### PR TITLE
test: audit bootstrap exports

### PR DESCRIPTION
**Pull request by dckc**
refs: #4548

## Description

First, admit you have a problem. :)

That is: before taking steps to minimize non-durable stuff exported from bootstrap, measure / audit it.

### Security / Documentation Considerations

none

### Scaling / Testing Considerations

Provides some scaling info by way of a test.

### CODE REVIEW

File: packages/vats/test/bootstrapTests/test-vaults-integration.js [hunk 1]
```diff
@@ -69,6 +69,110 @@ test.after.always(t => {
...
+    // makeAgoricNamesAccess()
+    'nameHub',
+    'nameAdmin',
+  ];
+
+  const { controller } = t.context;
+  const kState = controller.dump();
+  const bootstrapExports = kState.kernelTable.filter(
+    o => o[1] === 'v1' && o[2].startsWith('o+'),
+  );
```
Comment (mhofman) [inline packages/vats/test/bootstrapTests/test-vaults-integration.js]: Just thought about something. Should we be worried about bootstrap having some promises to which it remains the decider?

File: packages/vats/test/bootstrapTests/test-vaults-integration.js [hunk 2]
```diff
@@ -69,6 +69,110 @@ test.after.always(t => {
...
+
+  const exportedInterfaces = [...toIface.values()].map(iface =>
+    iface.replace(/^Alleged: /, ''),
+  );
+
+  // ava does a poor job of diffing string arrays
+  const extra = setDiff(exportedInterfaces, expectedIfaces);
+  t.deepEqual(extra, [], 'unexpected exported interfaces');
+  const missing = setDiff(expectedIfaces, exportedInterfaces);
+  t.deepEqual(missing, [], 'missing exported interfaces');
```
Comment (mhofman) [inline packages/vats/test/bootstrapTests/test-vaults-integration.js]: An alternative might be to make them a record with a `true` value or something. But yeah unfortunate.

**Review by mhofman — APPROVED** [review]
This all seems sensible

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 2 above)
Comment (turadg) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: That seems worthwhile to get a single assertion.

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 1 above)
Comment (dckc) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: well, now I am worried. Is it straightforward to find them in the kernel dump?

Comment (mhofman) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: Well for promises it's important to check who the current decider is.

I just quickly looked it up, and `kState.promises.filter(({decider}) => decider === 'v1')` should do the trick

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 2 above)
Comment (dckc) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: done in fixup!

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 1 above)
Comment (dckc) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: good news: there are none!

**Pull request merged**